### PR TITLE
* #10: renamed the project

### LINKED ISSUE #10: Fix the documentation and the examples after renaming the repository
## Problem

The repository was renamed from `go-exasol` to `exasol-driver-go`. We need to fix the documentation.

### PR DESCRIPTION
**Pull request by AnastasiiaSergienko**
Closes #10

### CODE REVIEW

File: dsn_test.go [hunk 1]
```diff
@@ -1,7 +1,6 @@
-package exasol_test
+package exasol
```
Comment (Nightapes) [inline dsn_test.go:1]: Why did you change the package?

Comment (AnastasiiaSergienko) [reply dsn_test.go:1]: I removed the import and want to access ParseDSN func without reference to `exasol`. Is it wrong?

Comment (Nightapes) [reply dsn_test.go:1]: With removing the `_test` we can now access internal code in the tests. Depends with we want to test this too or only the public methods.

File: examples/simple.go [hunk 2]
```diff
@@ -5,7 +5,7 @@ import (
 	"fmt"
 	"log"
 
-	_ "github.com/exasol/go-exasol"
+	_ "github.com/exasol/exasol-driver-go"
 )
 
 func main2() {
```
Comment (Nightapes) [inline examples/simple.go:11]: Maybe can you rename the method to `main`

Comment (AnastasiiaSergienko) [reply examples/simple.go:11]: I wanted to do it in the next pull request, I have a ticket for that: https://github.com/exasol/exasol-driver-go/issues/9

File: dsn_test.go (hunk 1 above)
Comment (AnastasiiaSergienko) [reply dsn_test.go:1]: I think it's fine for now as those are unit tests

Comment (AnastasiiaSergienko) [reply dsn_test.go:1]: It's also more consistent, all unit tests belong to `exasol` package now. for some of them I had to use private structures to test

**Review by Nightapes — APPROVED** [review]
